framework/test/unit/modules/pos/module.spec.ts

```diff
@@ -1139,10 +1140,58 @@ describe('PoS module', () => {
 
 				expect(result).toHaveLength(defaultConfig.numberActiveValidators);
 			});
+
+			it('should set averageWeight for initValidators', async () => {
+				context = createBlockContext({
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on stake weight to be sorted
+				const validators: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.validatorWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.validatorWeight),
+					})),
+				];
+				sortValidatorsByWeightDesc(validators);
+
+				const result = await pos['_getActiveValidators'](context, validators, 6);
+				expect(result).toHaveLength(defaultConfig.numberActiveValidators);
+				const notFromInitValidators = result.filter(
+					v => initValidators.findIndex(address => v.address.equals(address)) === -1,
+				);
+				const fromInitValidators = result.filter(
+					v => initValidators.findIndex(address => v.address.equals(address)) > -1,
+				);
+				const average =
+					notFromInitValidators.reduce((prev, curr) => prev + curr.weight, BigInt(0)) /
+					BigInt(notFromInitValidators.length);
+				expect(fromInitValidators.every(v => v.weight === average)).toBeTrue();
+			});
+
+			it('should scale BFT weight', async () => {
+				context = createBlockContext({
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on stake weight to be sorted
+				const validators: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.validatorWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.validatorWeight),
+					})),
+				];
+				sortValidatorsByWeightDesc(validators);
+
+				const result = await pos['_getActiveValidators'](context, validators, 6);
+				expect(result).toHaveLength(defaultConfig.numberActiveValidators);
+				const notFromInitValidators = result.filter(
+					v => initValidators.findIndex(address => v.address.equals(address)) === -1,
+				);
+				expect(notFromInitValidators.every(v => v.weight <= WEIGHT_SCALE_FACTOR)).toBeTrue();
+			});
 		});
 
 		describe('when current round is more than initRounds + numberOfActiveValidators', () => {
```

Comment:
"more or equal to `initRounds + numberOfActiveValidators`". Perhaps add an extra test when round is exactly `round = initRounds + numberOfActiveValidators` to check that we have the intended behavior.